salt/_states/metalk8s_kubernetes.py

```diff
@@ -1182,6 +1182,40 @@ def node_taints_present(name, taints, **kwargs):
     return ret
 
 
+def node_taints_absent(name, taints, **kwargs):
+    ret = {'name': name,
+           'changes': {},
+           'result': False,
+           'comment': ''
+    }
+    all_node_taints = __salt__['metalk8s_kubernetes.node_taints'](name, **kwargs)
+    existing_taints = {t['key']: t for t in all_node_taints}
+
+    for taint in taints:
+        key_to_remove = taint['key']
+        if key_to_remove in existing_taints:
+            existing_taints.pop(key_to_remove)
+
```

Comment:
You need to return here if taint already absent (I know it's not in `node_taints_present` but it's something missing)